design/rook-ceph-status-conditions.md

```diff
@@ -0,0 +1,93 @@
+
+## System States for rook-ceph
+
+The necessary system states for the rook-ceph can be portrayed as follows:
+	   
+	   Ignored 		: If any of the resources gets ignored for multiple reasons
+	   Progressing 		: Marks the start of reconcile of Ceph Cluster
+	   Ready 		: When Reconcile completes successfully
+	   Not Ready 		: Either when cluster is Upgrading or Upgrading is blocked
+	   Connecting		: When the Ceph Cluster is in the state of Connecting 
+	   Available 		: The Ceph Cluster is healthy and is available to use
+	   Failure 		: If any failure occurs in the Ceph Cluster
+	   Cluster Expanding	: If the Cluster is Expanding
+
+## Implementation Details
+
+Implementation as Specified in https://github.com/openshift/custom-resource-status/:
+
+We can add Conditions simply in the Custom Resource struct as:
+	  
+	   type ClusterStatus struct{
+		Conditions     *RookConditions    `json:"conditions,omitempty"`
+		CephStatus     *CephStatus        `json:"ceph,omitempty"`
+	   }
+
+After that we can just make changes inside rook ceph codebase as necessary. 
+	   
+The definition of the type Conditions can have the following details:
+	   
+	   Type               RookConditionType  `json:"type" description:"type of Rook condition"`
+  	   Status             ConditionStatus    `json:"status" description:"status of the condition, one of True, False, Unknown"`
+  	   Reason             *string            `json:"reason,omitempty" description:"one-word CamelCase reason for the condition's last transition"`
+  	   Message            *string            `json:"message,omitempty" description:"human-readable message indicating details about last transition"`
+	   LastHeartbeatTime  *unversioned.Time  `json:"lastHeartbeatTime,omitempty" description:"last time we got an update on a given condition"`
+	   LastTransitionTime *unversioned.Time  `json:"lastTransitionTime,omitempty" description:"last time the condition transition from one status to another"`
+
+The fields `Reason`, `Message`, `LastHeartbeatTime`, `LastTransitionTime` are optional field. Though the use of `Reason` field is encouraged.
+
+Condition Types field specifies the current state of the system. Condition status values may be `True`, `False`, or `Unknown`. The absence of a condition should be interpreted the same as Unknown. How controllers handle Unknown depends on the Condition in question.
+`Reason` is intended to be a one-word, CamelCase representation of the category of cause of the current status, and `Message` is intended to be a human-readable phrase or sentence, which may contain specific details of the individual occurrence. `Reason` is intended to be used in concise output, such as one-line kubectl get output, and in summarizing occurrences of causes, whereas `Message` is intended to be presented to users in detailed status explanations, such as `kubectl describe output`.
```

Comment:
Can we also address this with an example and brief implementation steps? 
Personally, to me it looks like a key piece. Why ask the user to dig deeper when we can surface out the key information.

Reply:
We can show the key user with a message field along with the readable status. I included an example in the updated design doc